ButtonGroup Component

antictor opened the pull request:
## Related issue
#219

The button group accepts a list of button item properties and renders a group of like styled buttons as a unit.

![Oct-30-2019 10-10-31](https://user-images.githubusercontent.com/48056118/67849002-a1d0db00-fafd-11e9-91bb-9a6ca2547b67.gif)

antictor committed:
Update button group styles

antictor committed:
Add ButtonGroup React component

antictor committed:
Add ButtonGroup developer documentation

thyhjwb6 reviewed (commented):
A few small comments

antictor committed:
Change button group behaviour

Changes the ButtonGroup component to generate markup for a collection of buttons, grouped together. All the properties of the buttons are set at the top level, and filter down and buttons are now only secondary buttons with no colour variation. The ability to include disabled buttons has also been added.

antictor committed:
Add test id to button to help with tests

antictor committed:
Improve routing onClick in ButtonGroup and added tests

Simplified ButtonGroup a little by moving the onClick for buttons into the individual button items.

Created unit tests for ButtonGroup

antictor committed:
Update ButtonGroup developer documentation

Updated the documentation for this component to take into account the changes to it's structure and way of working.

antictor committed:
Updated button group highlight to use primary border colour

hxltrhuxze committed:
Button Group design docs
